### ISSUE TITLE
[Bug] Does not work with RoleManager

### ISSUE THREAD

**Issue by dougwaldron** [AzureAD/microsoft-identity-web#1091]
As described in the [Azure Architecture documentation](https://docs.microsoft.com/en-us/azure/architecture/multitenant-identity/app-roles#roles-using-an-application-role-manager), one strategy for defining  roles is to store role assignments in the application's own DB, "for example, using the RoleManager class in ASP.NET Identity." But the docs give no implementation instructions. I've tried every way I can think of, and I can't get Microsoft Identity Web to work with RoleManager.

We have an app in production that uses [Microsoft.AspNetCore.Authentication.AzureAD.UI](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.AzureAD.UI) for authentication, with roles managed by the app. The latest version of this package comes with a warning that [`AddAzureAD` and related methods are obsolete](https://github.com/dotnet/aspnetcore/blob/e882394a7bb38930da748291fe2c2ceaa6a80588/src/Azure/AzureAD/Authentication.AzureAD.UI/src/AzureADAuthenticationBuilderExtensions.cs#L49) and "will be removed in a future version." If this happens and Microsoft Identity Web can't be made to work with RoleManager, then our app will break and we'll have to stop using Azure AD for authentication.

This is a duplicate of an earlier issue - #680 - but that issue was closed without a satisfactory resolution because the OP moved on to a different strategy. In that issue, you recommended defining roles within Azure AD instead. Besides our specific situation described above, I would argue that authentication and role management are two separate functions and shouldn't be relied on to exist in one service.

**Comment by dougwaldron** [comment]
I returned to this problem last month and found that I was able to migrate my apps from AzureAD.UI to Microsoft.Identity.Web and also continue to use ASP.NET Identity's RoleManager to manager application roles. 

Since it's been so long since I initially opened this issue, I can't remember exactly what I had tried before, so I can't say if your plugin changed somehow in the meantime, of if I was able to find a workaround. Either way, this issue is resolved for me. Thanks!
